TodoApp.Repositories/XmlRepository/RepositoryBase.cs

```diff
@@ -1,20 +1,19 @@
 ﻿using System;
 using System.Collections.Generic;
 using System.Xml.Linq;
-using TodoApp.ConsoleApp.Repositories.Interfaces;
-using TodoApp.ConsoleApp.Repositories.Models;
+using TodoApp.Repositories.Interfaces;
+using TodoApp.Repositories.Models;
+using TodoApp.Repositories.XmlRepository.Utils;
 using System.Linq;
-using System.IO;
 using System.Data;
 
-namespace TodoApp.ConsoleApp.Repositories.XmlRepository
+namespace TodoApp.Repositories.XmlRepository
 {
     public abstract class RepositoryBase<TModel, TId> : IRepository<TModel, TId> where TModel : IModel<TId>
     {
-        protected string Path { get; set; }
-        protected XDocument Document { get; }
         protected XElement ContainerElement { get; }
         protected abstract string IdName { get; }
+        protected IXmlContext Context;
 
         protected XElement GetElementById(TId id)
```

Comment:
The methods should be defined **after** the constructor.
Maybe we have missed this on the previous PR.